### PR TITLE
New Document Type and Properties format

### PR DESCRIPTION
**Pull request by KshChange**
### Issue being fixed or implemented
- New Document Type and Properties regexp

### What was done
- Changed regexp in contract schema
- Added tests

### CODE REVIEW

File: schema/meta/contract.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
       "type": "string",
       "minLength": 3,
       "maxLength": 24,
-      "pattern": "^[a-zA-Z0-9-_]{1,255}$"
+      "pattern": "^((?!-)[a-zA-Z0-9-]{0,62}[a-zA-Z0-9])$"
```
Comment (jawid-h) [inline schema/meta/contract.json]: Please, increase number of characters to 255 by using `{0,254}` instead of `{0,62}`

**Comment by jawid-h** [comment]
Please, update version number. Probably minor version since it's not breaking backward compatibility. @shumkov, please confirm.

File: schema/meta/contract.json (hunk 1 above)
Comment (jawid-h) [reply schema/meta/contract.json]: You should change it in other places as well.

File: test/integration/contract/validateContractFactory.spec.js [hunk 2]
```diff
@@ -305,17 +351,34 @@ describe('validateContractFactory', () => {
       expect(error.keyword).to.equal('minProperties');
     });
 
-    it('should have no non-alphanumeric properties', () => {
-      rawContract.documents['(*&^'] = rawContract.documents.niceDocument;
+    it('should be a valid string', () => {
```
Comment (shumkov) [inline test/integration/contract/validateContractFactory.spec.js]: should have valid property names

File: test/integration/contract/validateContractFactory.spec.js [hunk 3]
```diff
@@ -123,17 +123,33 @@ describe('validateContractFactory', () => {
       expect(error.keyword).to.equal('maxLength');
     });
 
-    it('should be an alphanumeric string', () => {
-      rawContract.name = '*(*&^';
+    it('should be a valid string', () => {
```
Comment (shumkov) [inline test/integration/contract/validateContractFactory.spec.js]: should be the valid string format

File: test/integration/contract/validateContractFactory.spec.js [hunk 4]
```diff
@@ -123,17 +123,33 @@ describe('validateContractFactory', () => {
...
+        const result = validateContract(rawContract);
 
-      const [error] = result.getErrors();
+        expectJsonSchemaError(result, 0);
+      });
+    });
 
-      expect(error.dataPath).to.equal('.name');
-      expect(error.keyword).to.equal('pattern');
+    it('should not pass any invalid string', () => {
```
Comment (shumkov) [inline test/integration/contract/validateContractFactory.spec.js]: should return an invalid result if a string is in invalid format

**Pull request merged**